pkgs/development/python-modules/picos/default.nix

```diff
@@ -1,41 +1,48 @@
 {
   lib,
   buildPythonPackage,
-  fetchFromGitLab,
+  fetchPypi,
   numpy,
   cvxopt,
   python,
   networkx,
+  scipy,
+  pythonOlder,
 }:
 
 buildPythonPackage rec {
   pname = "picos";
-  version = "2.0";
+  version = "2.5.1";
   format = "setuptools";
 
-  src = fetchFromGitLab {
-    owner = "picos-api";
-    repo = "picos";
-    rev = "v${version}";
-    sha256 = "1k65iq791k5r08gh2kc6iz0xw1wyzqik19j6iam8ip732r7jm607";
+  disabled = pythonOlder "3.12";
```

Comment:
According to the readme it should be compatible with Python 3.4 and older:
> As of release 2.2, PICOS requires Python 3.4 or later.

I think it would be better to disable the failing test.